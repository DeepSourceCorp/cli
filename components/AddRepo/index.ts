import AddRepoModal from './AddRepoModal.vue'
import RepoCard from './RepoCard.vue'
import SyncRepoAlert from './SyncRepoAlert.vue'

export { AddRepoModal, RepoCard, SyncRepoAlert }
