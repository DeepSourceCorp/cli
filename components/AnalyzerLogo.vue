<template>
  <div>
    <img
      v-if="analyzerLogo"
      v-tooltip="hideTooltip ? '' : name"
      class="w-auto flex-shrink-0"
      :class="sizeStyle"
      :src="analyzerLogo"
      :alt="name"
    />
    <z-icon
      v-else
      v-tooltip="hideTooltip ? '' : name"
      :icon="shortcode"
      color="transparent"
      :size="size"
    />
  </div>
</template>
<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator'

@Component({})
export default class AnalyzerLogo extends Vue {
  @Prop({ default: '' })
  analyzerLogo!: string

  @Prop({ default: '' })
  name!: string

  @Prop({ default: '' })
  shortcode!: string

  @Prop({ default: 'base' })
  size!: string

  @Prop({ default: false })
  hideTooltip: boolean

  get sizeStyle() {
    const sizes: Record<string, string> = {
      'x-small': 'w-3 h-3',
      small: 'w-4 h-4',
      base: 'w-5 h-5',
      medium: 'w-6 h-6',
      large: 'w-8 h-8'
    }

    return sizes[this.size] || 'w-5 h-4'
  }
}
</script>
