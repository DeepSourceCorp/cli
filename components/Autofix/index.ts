import InfoIcon from './InfoIcon.vue'
import Info from './Info.vue'
import AutofixCard from './AutofixCard.vue'
import AutofixListItem from './AutofixListItem.vue'
import AutofixCodeDiff from './AutofixCodeDiff.vue'

export { InfoIcon, AutofixCard, AutofixListItem, AutofixCodeDiff, Info }
