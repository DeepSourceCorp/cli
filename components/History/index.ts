import BaseCard from './BaseCard.vue'
import BranchList from './BranchList.vue'
import SubNav from './SubNav.vue'

import RunBranches from './Runs/RunBranches.vue'
import RunCard from './Runs/RunCard.vue'

import TransformBranches from './Transforms/TransformBranches.vue'
import TransformCard from './Transforms/TransformCard.vue'

export { RunBranches, RunCard, BranchList, TransformBranches, TransformCard, BaseCard, SubNav }
