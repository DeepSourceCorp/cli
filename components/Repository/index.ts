import GistCard from './GistCard.vue'
import GistCardTitle from './GistCardTitle.vue'
import GistCardDescription from './GistCardDescription.vue'
import IssueType from './IssueType.vue'
import IgnoredRule from './IgnoredRule.vue'
import IssueSeverityTag from './IssueSeverityTag.vue'

export { GistCard, GistCardTitle, GistCardDescription, IssueType, IgnoredRule, IssueSeverityTag }
