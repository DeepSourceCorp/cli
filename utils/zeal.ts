export const toPascal = (word: string) => {
  // https://stackoverflow.com/a/53952925
  return `${word}`
    .replace(new RegExp(/[-_]+/, 'g'), ' ')
    .replace(new RegExp(/[^\w\s]/, 'g'), '')
    .replace(
      new RegExp(/\s+(.)(\w+)/, 'g'),
      ($1, $2, $3) => `${$2.toUpperCase() + $3.toLowerCase()}`
    )
    .replace(new RegExp(/\s/, 'g'), '')
    .replace(new RegExp(/\w/), (s) => s.toUpperCase())
}

export const containsElement = (parentCandidate: HTMLElement, target: HTMLElement): boolean => {
  return Boolean(
    parentCandidate && (target === parentCandidate || parentCandidate.contains(target))
  )
}
