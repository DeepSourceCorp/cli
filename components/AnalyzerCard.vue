<template>
  <div class="relative h-72 min-h-1/2 md:h-40 bg-ink-400 p-2 rounded-sm flex flex-col space-y-3 md:min-w-1/2 md:w-1/2 box-border overflow-scroll"
        :class="{ 'justify-center items-center space-y-2': configList.length }">
    <!-- Header -->
    <div v-if="configList.length > 0" class="flex w-full items-center space-x-2">
        <z-icon :icon="icon" size="small" color="robin"></z-icon>
        <!-- TODO: Get Repo name and handle name -->
        <div class="flex-1 text-sm text-vanilla-200 font-bold">
            {{ name }}
        </div>
        <div class="cursor-pointer" @click="handleClick">
            <z-icon icon="x"
                size="small" 
                color="vanilla-200">
            </z-icon>
        </div>
    </div>
    <!-- Config Section -->
    <template v-for="config in configList">
        <div :key="config.title" class="flex flex-col space-y-2">
            <div class="text-xs tracking-wide text-slate uppercase">
                {{config.title}}
            </div>
            <div v-if="config.type == 'boolean'">
                <z-radio-group v-model="config.selected" class="flex space-x-4 text-xs">
                    <z-radio v-for="option in config.values" 
                            :key="option.value"
                            :value="option.value"
                            :label="option.label"
                            ></z-radio>
                </z-radio-group>
            </div>
            <div v-else-if="config.type == 'input'">
                <z-input v-model="config.selected" 
                        textSize="xs">
                </z-input>
            </div>
            <div v-else-if="config.type == 'list'" class="h-8">
                <!-- TODO: Customizable placeholder -->
                <z-select 
                    v-model="config.selected" 
                    class="z-10" 
                    placeholder="Choose a framework">
                    <z-option
                        v-for="item in config.values"
                        :key="item.value"
                        :label="item.label"
                        :value="item.value">
                        <div class="flex items-center">
                            <z-icon icon="activity" size="small" class="-mt-1 mr-2"></z-icon>
                            <span class="flex-1">{{item.label}}</span>
                        </div>
                    </z-option>
                </z-select>
            </div>
        </div>
    </template>
    <!-- No Config Data Section -->
    <template v-if="configList.length == 0">
        <div class="absolute top-2 right-2 cursor-pointer" @click="handleClick">
            <z-icon icon="x"
                size="small" 
                color="vanilla-200">
            </z-icon>
        </div>
        <z-icon :icon="icon" color="pink"></z-icon>
        <div class="font-bold text-vanilla-100 text-sm">{{name}}</div>
    </template>
  </div>
</template>

<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator';
import { ZIcon, ZInput, ZSelect, ZOption, ZRadioGroup, ZRadio } from '@deepsourcelabs/zeal';
@Component({
    components: {
       ZIcon,
       ZInput,
       ZSelect,
       ZOption,
       ZRadioGroup,
       ZRadio
    }
})
export default class AnalyzerCard extends Vue {
    @Prop({default: 'lock'})
    icon!: string
    @Prop({default: ''})
    name!: string
    @Prop({default: []})
    configList!: Array<Record<string, unknown>>
    public handleClick(event: Event): void {
        this.$emit('click', event)
    }
}
</script>
