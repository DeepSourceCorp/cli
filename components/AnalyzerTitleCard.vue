<template>
  <div class="px-2 py-2 bg-ink-100 rounded-sm bg-ink-400 flex items-center space-x-2 cursor-pointer"
        @click="handleClick()">
    <z-icon :icon="icon" size="small" color="honey"></z-icon>
    <div class="flex-1 text-sm text-vanilla-200 font-bold">{{name}}</div>
    <z-icon icon="plus" size="small" class="cursor-pointer" color="vanilla-200"></z-icon>
  </div>
</template>

<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator';
import { ZIcon } from '@deepsourcelabs/zeal';
@Component({
    components: {
       ZIcon 
    }
})
export default class AnalyzerTitleCard extends Vue {
    @Prop({default: 'globe'})
    icon!: string
    @Prop({default: ''})
    name!: string
    public handleClick(event: Event): void {
        this.$emit('click', event)
    }
}
</script>
