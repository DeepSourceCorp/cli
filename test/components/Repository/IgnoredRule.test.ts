import { render } from '@testing-library/vue'
import { IgnoredRule } from '~/components/Repository'
import { BaseCard } from '~/components/History'
import { ZIcon, ZButton } from '@deepsource/zeal'
import { VTooltip } from 'v-tooltip'
import { VueConstructor } from 'vue'
import { RouterLinkStub } from '@vue/test-utils'

const injectDirective = (vue: VueConstructor) => vue.directive('tooltip', VTooltip)

test('renders IgnoredRule with all prop options', () => {
  const props = {
    rule: {
      silenceLevel: 'RL',
      id: 'U2lsZW5jZVJ1bGU6eXp5bHZi',
      filePath: null,
      createdAt: '2022-03-14T11:50:48.350324+00:00',
      metadata: { type: 'test-pattern' },
      issue: {
        shortcode: 'PYL-W0621',
        title: 'Re-defined variable from outer scope',
        analyzer: { shortcode: 'python' }
      },
      creator: {
        firstName: 'Vaibhav',
        lastName: '',
        email: 'vaibhav@deepsource.io',
        avatar: '/static/dashboard/images/empty-avatar.svg'
      }
    },
    verticalPadding: 'pb-4'
  }

  const mocks = {
    $route: {
      query: ''
    },
    viewer: {},
    $config: { onPrem: false },
    $socket: {
      $on: () => {},
      $off: () => {}
    },
    async $fetchGraphqlData() {}
  }

  const { html } = render(
    IgnoredRule,
    {
      props,
      mocks,
      stubs: {
        NuxtLink: RouterLinkStub,
        BaseCard: BaseCard,
        ZIcon: true,
        ZButton: true
      },
      computed: {
        allowDelete() {
          return true
        },
        ruleCreatedTime() {
          return '30 minutes ago'
        }
      }
    },
    injectDirective
  )

  expect(html()).toMatchSnapshot(JSON.stringify(props))
})
