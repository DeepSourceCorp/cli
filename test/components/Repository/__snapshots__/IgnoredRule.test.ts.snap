// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders IgnoredRule with all prop options: {"rule":{"silenceLevel":"RL","id":"U2lsZW5jZVJ1bGU6eXp5bHZi","filePath":null,"createdAt":"2022-03-14T11:50:48.350324+00:00","metadata":{"type":"test-pattern"},"issue":{"shortcode":"PYL-W0621","title":"Re-defined variable from outer scope","analyzer":{"shortcode":"python"}},"creator":{"firstName":"Vaibhav","lastName":"","email":"vaibhav@deepsource.io","avatar":"/static/dashboard/images/empty-avatar.svg"}},"verticalPadding":"pb-4"} 1`] = `
"<div to=\\"\\" class=\\"group block\\">
  <!---->
  <div class=\\"flex w-full flex-1 flex-wrap\\">
    <div class=\\"space-y-1 border-slate-400 text-sm text-vanilla-400 w-full md:w-4/5 pb-4\\">
      <div class=\\"flex items-center space-x-2 text-base font-semibold text-vanilla-200\\">
        <p class=\\"text-sm font-normal\\"><a target=\\"_blank\\" rel=\\"noopener noreferrer\\"><span class=\\"font-bold text-juniper hover:underline\\">PYL-W0621</span></a> <span>Ignored</span> <span>
        for all test files in the repository
      </span> <span class=\\"font-semibold text-vanilla-100\\"></span></p>
      </div>
      <div class=\\"flex gap-x-4 text-xs\\"><span class=\\"flex items-center gap-x-1\\"><img src=\\"/static/dashboard/images/empty-avatar.svg\\" alt=\\"Creator Avatar\\" class=\\"inline-block h-4 w-4 overflow-hidden rounded-full\\"> <span class=\\"text-vanilla-400\\">vaibhav@deepsource.io</span></span> <span class=\\"flex items-center gap-x-2 leading-none\\"><zicon-stub icon=\\"clock\\" color=\\"vanilla-400\\" size=\\"small\\"></zicon-stub> <span class=\\"text-vanilla-400\\">Added 30 minutes ago</span></span></div>
    </div>
    <div class=\\"hidden w-full md:block md:w-1/5\\">
      <div class=\\"mt-1 flex h-full justify-end\\">
        <zbutton-stub size=\\"small\\" button-type=\\"ghost\\" icon=\\"trash-2\\" color=\\"cherry\\" class=\\"v-popper--has-tooltip\\"></zbutton-stub>
      </div>
    </div>
  </div>
</div>"
`;
